server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -203,6 +203,14 @@ public List<String> dimensions() {
         return Collections.emptyList();
     }
 
+    public String getInferenceModel() {
```

Comment:
Method added for future retrieval from `MappingLookup`

Reply:
@carlosdelest I would prefer a new interface & `instanceof` to check that something satisfies this new API. I don't like the idea of having it at the base MappedFieldType.

Reply:
I am unclear on where the method would be called / where would the instanceof check go. Could you clarify?

Reply:
I agree on the sentiment, I just could not see what the impact is of an instanceof because the method is currently only exposed but never used. It is also true that this kind of special cases complicate the code in the long run, but I see what you mean.

Reply:
@javanna I think @carlosdelest is trying to lay the ground work for gathering field capabilities in some way to determine if a model is used and what model. 

I agree, it's way too early here to make any changes to wide expansive interfaces like this one.

Reply:
As Ben said, this will be used to gather the fields that are using an ML model so they can be included in `IndexMetadata` in a later stage.

I'm OK with using a new interface and defer the decision to include this method as part of the base class. 

@Mikep86, please be aware that I'll be moving `getInferenceModel()` from `MappedFieldType` to a new inteface.

Reply:
Addressed in 048779b7830843a43954e96b3ddc832b0cfb1ddd